From: Arend van Beelen jr. <arend@fiberplane.com>
Date: Tue, 29 Sep 2022 16:37:00 +0200
Subject: [PATCH] Manually edit OpenAPI generated code

The OpenAPI client generator for Rust does not handle untagged enums well.
So instead, we patch in some code based on our `fiberplane-rs` types.

---
 fiberplane-api-client/src/models/new_time_range.rs | 18 +-
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/fiberplane-api-client/src/models/new_time_range.rs b/fiberplane-api-client/src/models/new_time_range.rs
index a6b7d55..e1a5b77 100644
--- a/fiberplane-api-client/src/models/new_time_range.rs
+++ b/fiberplane-api-client/src/models/new_time_range.rs
@@ -9,17 +9,13 @@
  */

 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
-pub struct NewTimeRange {
-    #[serde(rename = "from")]
-    pub from: String,
-    #[serde(rename = "to")]
-    pub to: String,
-    #[serde(rename = "relative")]
-    pub relative: i32,
+#[serde(untagged)]
+pub enum NewTimeRange {
+    Absolute(crate::models::TimeRange),
+    Relative(RelativeTimeRange),
 }

-impl NewTimeRange {
-    pub fn new(from: String, to: String, relative: i32) -> NewTimeRange {
-        NewTimeRange { from, to, relative }
-    }
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct RelativeTimeRange {
+    pub minutes: i32,
 }
